web/gds-user-ui/src/components/Modal/ConfirmationResetFormModal.tsx

```diff
@@ -12,73 +12,108 @@ import {
+    useDeleteCertificateStep(step);
+  const { updateStepStatusState } = useCertificateStepper();
+  // const navigate = useNavigate();
   const { onClose: onAlertClose } = useDisclosure();
-  const { resetForm } = useCertificateStepper();
-  const [isLoading, setIsLoading] = useState<boolean>(false);
+  // const { resetForm } = useCertificateStepper();
+
   const handleOnClose = () => {
     props.onClose();
     onAlertClose();
-    props.onChangeState(false);
+    // props.onChangeState(false);
   };
+
+  const isResetAllType = props.step === StepEnum.ALL;
+
   const handleResetBtn = () => {
-    setIsLoading(true);
-    // props.onReset(loadDefaultValueFromLocalStorage);
-    resetForm();
-    props.onReset(getRegistrationDefaultValues());
-    props.onChangeResetState(true);
-    props.onChangeState(false);
-    setIsLoading(false);
-    props.onClose();
-    onAlertClose();
-    // props.onRefreshState();
-    navigate('/dashboard/certificate/registration');
+    deleteCertificateStep({
+      key: props.step as StepperType
+    });
+    // resetForm();
+    // props.onChangeResetState(true);
+    // props.onChangeState(false);
+
+    // navigate('/dashboard/certificate/registration');
   };
+
+  useEffect(() => {
+    if (wasCertificateStepDeleted) {
+      updateStepStatusState({
+        step: getStepNumber(props.step),
+        status: 'incomplete'
+      });
+      props.onClose();
+    }
+  }, [wasCertificateStepDeleted, props, onAlertClose, updateStepStatusState]);
+
   return (
     <>
       <Flex>
         <Box w="full">
-          <Modal closeOnOverlayClick={false} {...props}>
+          <Modal closeOnOverlayClick={false} {...props} isOpen={props.isOpen}>
             <ModalOverlay />
             <ModalContent width={'100%'}>
               <ModalHeader data-testid="confirmation-modal-header" textAlign={'center'}>
-                <Trans id="Clear & Reset Registration Form">Clear & Reset Registration Form</Trans>
+                {isResetAllType ? (
+                  <Trans>Clear & Reset Registration Form</Trans>
+                ) : (
+                  <Trans>Clear & Reset Section</Trans>
+                )}
               </ModalHeader>
 
               <ModalBody pb={5}>
                 <Text pb={2} fontSize={'sm'}>
-                  <Trans id="Click “Reset” to clear and reset the registration form. All data will be deleted and you will be re-directed to the beginning of the form and you will be required to restart the registration process">
-                    Click “Reset” to clear and reset the registration form. All data will be deleted
-                    and you will be re-directed to the beginning of the form and you will be
-                    required to restart the registration process
-                  </Trans>
+                  {isResetAllType ? (
+                    <Trans>
+                      Click “Reset” to clear and reset the registration form. All data will be
```

Comment:
```suggestion
                      Click "Reset" to clear and 
                      <Text as="span" fontWeight={'bold'}>
                        {' '}
                        reset the form{' '}
                      </Text>{' '}. All data will be
```

Reply:
Modal changes in Figma - https://www.figma.com/file/fHC69EVhfNtsKef1yKgW33/TRISA_GDS_VASP_UI?type=design&node-id=6010-2045&t=R7dZEeksnyIP6dVL-4